qcodes/instrument_drivers/Spectrum/M4i.py

```diff
@@ -777,22 +786,41 @@ def start_acquisition(self, mV_range, memsize, posttrigger_size=None, verbose=0)
 
         return {'memsize': memsize, 'numch': numch, 'mV_range': mV_range}
 
-    def _transfer_buffer_numpy(self, memsize, numch):
-        """ Transfer buffer to numpy array """
+    def _transfer_buffer_numpy(self, memsize, numch, bytes_per_sample=2):
+        """ Transfer buffer to numpy array
+
+        Args:
+            memsize (int): number of samples to transfer
+            numch (int): number of channels
+            bytes_per_sample (int): specifies the datatype. 2 for int16, 4 for int32
+        Returns:
+            array: transfered data
+
+        """
         # setup software buffer
-        buffer_size = ct.c_int16 * memsize * numch
+        if bytes_per_sample == 2:
+            buffer_size = ct.c_int16 * memsize * numch
+        elif bytes_per_sample == 4:

```

Comment:
i'd highly recommend to have the if-clause for bytes_per_sample only once in the code. Basically, bytes_per_sample define what the ctype type is going to be used. Once the ctype is determined, it can be used for buffer_size and for dtype in np.from_buffer without the need for repeating the if-clause. This make the code shorter, more readable, and reduces the possibility for bugs and typos.